docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+  - MJML can be integrated with React to generate responsive emails and there is also a [library](https://github.com/wix-incubator/mjml-react) that could help us with that purpose.
+- Cons
+  - Adds a new dependency to the stack
+  - Creates a slight learning curve
+  - While we may not have much component reuse between other packages that use React anyway, using MJML in email components would make it more challenging to reuse them elsewhere
+
+## Pros and Cons of the Styling Options
+
+When emails are built by the server, we need the styles to be inserted inline since most email clients disregard style tags. However, inline styles aren't great for consistency across the look and feel of our emails and they make maintenance more difficult. We want to pick a solution that lets us use reusable classes during development but when built, can be turned into inline styles
+
+### Option A - Continue to use adhoc inline styling
+
+- Description
+  - Keep the current inline styling as is in the templates.
+- Pros
+  - Since most of the email clients work well with inline styling, the effort required to investigate which templates are not being rendered correctly would be minimized to a great extent.
+- Cons
+  - Styles added inline with the HTML tags can hamper the look and feel of the templates.
+  - Reusability of the styles cannot be expected, thereby making the emails inconsistent.
+  - Redundancy across the templates would be increased since the tags cannot be reused.
+  - It becomes almost impossible to use media queries with inline styles
+
+### Option B - Use Tailwind CSS
+
+- Description
+  - Most of the email clients disregard style tags while sending over emails so it is important for us to choose a styling framework that will generate compiled inline styles at build step and make reusing the style components possible across the templates. [TailwindCSS](https://tailwindcss.com/docs), with its utility-based approach, can help us create utility classes that can be reused and repurposed during development.
+- Pros
+  - TailwindCSS is a utility-first open source framework with ample documentation, active community, and immense user base which could be helpful to receive help and support.
+  - Tailwind's classes can be applied directly to the HTML elements eliminating the need of writing stylesheets.
+  - We can easily extract out duplicate style components using Tailwind maintaining consistency across our emails.
+  - It is already being used across fxa like in `fxa-payments-server`
+  - Tailwind classes can be compiled down into inline styles with the help of [juice](https://github.com/Automattic/juice) library.
+  - TailwindCSS provides us the opportunity to work with single-purpose classes as well as component classes which can help DRY up our codebase.
+  - Tailwind is highly configurable without being cumbersome, allowing us to modify type and spacing scales, define color ranges, and set up media queries to meet our exact needs.
+- Cons
```

Comment:
Would be worth noting that using Tailwind also means that we are left to re-implement all the various email client CSS hacks needed for proper rendering.

Reply:
Looks like there's [Maizzle](https://maizzle.com/) to pair with Tailwind as an email-specific framework. Not sure how that compares to MJML, but I've seen them mentioned in the same breath a lot as alternatives.

Reply:
I agree that could be listed as a con. I figured since the styles are already in-place it wouldn't be too painful to replace them with Tailwind classes, but unless we used Maizzle with Tailwind, we would have to maintain some of the hacky maintenance burden. 🤔

I think @Yogita98 mentioned Maizzle once before in Matrix. Have you poked at that out of curiosity? 🙂